py4DSTEM/process/calibration/ellipse.py

```diff
@@ -602,4 +604,99 @@ def compare_double_sided_gaussian(data, p, power=1, mask=None):
+
+    Returns: 
+    background1D (ndarray) 1D polar elliptical background
+    r_bins       (ndarray) the elliptical semimajor axis radius associated with background1D
+    PolarData    (ndarray) **optional: returns the masked polar transform from which the 1D background is computed
+    """
+    # Compute Polar Transform
+    PolarData, rr, tt = cartesianDataAr_to_polarEllipticalDataAr(data,tuple([qx0,qy0, 1, e, phi]))
+   
+    # Crop polar data to maximum distance which contains information from original image
+    if (PolarData.mask.sum(axis = (0))==PolarData.shape[0]).any():
+            Maximal_Distance  = np.min(np.where(PolarData.mask.sum(axis = (0))==PolarData.shape[0]))
+            PolarData = PolarData[:,0:Maximal_Distance]
+            r_bins = rr[0,0:Maximal_Distance]
+    else:
+            r_bins = rr[0,:]
+
+    # Iteratively mask off high intensity peaks
+    Mask_Polar = PolarData.mask
+    for ii in range(mask_update_iter+1): 
+        if ii > 0:    
+            mask_update = np.logical_or(Mask_Polar,  
+                                        PolarData/background_1D > min_relative_threshold)
+            # Prevent entire columns from being masked off 
+            col_mask_min = np.all(mask_update, axis = 0)  # Detect columns that are empty
+            mask_update[:,col_mask_min] = PolarData.mask[:,col_mask_min] # reset empty columns to values of previous iterations
+            PolarData.mask  = mask_update  # Update Mask
+
+        background_1D = np.ma.median(PolarData, axis = 0) 
+
+    background_1D = np.maximum(background_1D, min_background_value)
+
+    if smoothing_window_size != None:
+        if smoothing_log==True: 
+            background_1D = np.log(background_1D)
+
+        background_1D = savgol_filter(background_1D, 
+                                     smoothing_window_size, 
+                                     smoothing_poly_order)
+        if smoothing_log==True: 
+            background_1D = np.exp(background_1D)
+    if return_polararr ==True: 
+        return(background_1D, r_bins, PolarData)
+    else: 
+        return(background_1D, r_bins)
+
+#Create 2D Background 
+def get_2Dbackground(data, background1D, r_bins, qx0, qy0, phi, e):
+    """
+    Gets 2D polar elliptical background from linear 1D background 
+
+    Accepts:
+    data            (ndarray) the data for which to find the polar eliptical background, usually a diffraction pattern
+    background1D:   (ndarray) a vector representing the radial elliptical background 
+    r_bins:         (ndarray) a vector of the elliptical semimajor axis radius associated with background1D
+    qx0,qy0         (numbers) the ellipse center; if the braggpeaks have been centered,
+                              these should be zero
+    e               (number) the length ratio of semiminor/semimajor axes
+    phi             (number) tilt of the major axis with respect  to (qx, qy) axis, in radians 
+
+    Returns: 2D Background
+    """
+    
```

Comment:
add an assert statement to ensure correct dimensions for `r_bins` and `background1d`